firedrake/plot.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import absolute_import
+import numpy as np
+from ufl import cell
+from firedrake import Function, SpatialCoordinate
+
+__all__ = ["plot"]
+
+
+def _plot_mult(functions, num_points=100):
+    """Plot multiple functions on a figure, return a matplotlib figure
+    :arg functions: Functions to be plotted
+    :arg num_points: Number of points per element
+    """
+    try:
+        import matplotlib.pyplot as plt
+        from matplotlib.widgets import Slider
+    except ImportError:
+        raise RuntimeError("Matplotlib not importable, is it installed?")
+    if len(functions) == 0:
+        return None
+    figure, ax = plt.subplots()
+    func_axis = plt.axes([0.25, 0.025, 0.65, 0.03], axisbg='lightgoldenrodyellow')
+    func_slider = Slider(func_axis, "Func Select", 0.1, len(functions), valinit=0)
+    func_slider.valtext.set_text('0')
+
+    def update(val):
+        val = int(val - 0.1)
+        func_slider.valtext.set_text('{:.0f}'.format(val))
+        ax.cla()
+        plot(functions[val], ax)
+    update(0)
+    func_slider.on_changed(update)
+    return figure
+
+
+def plot(function, axes=None, num_points=100, **kwargs):
+    """Plot a function or a list of functions and return a matplotlib
+    figure object.
+    :arg function: The function to plot.
+    :arg axes: Axes to be plotted on
+    :arg num_points: Number of points per element
+    :arg kwargs: Additional keyword arguments passed to
+    ``matplotlib.plot``.
+    """
+
+    if not isinstance(function, Function):
+        return _plot_mult(function, num_points)
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise RuntimeError("Matplotlib not importable, is it installed?")
+    if function.function_space().mesh().ufl_cell() == cell.Cell("interval"):
+        if function.function_space().ufl_element().degree() < 4:
+            return bezier_plot(function, axes, **kwargs)
+        points = calculate_one_dim_points(function, num_points)
+        if axes is None:
+            axes = plt.subplot(111)
+        axes.plot(points[0], points[1], **kwargs)
+        return plt.gcf()
+    else:
+        raise RuntimeError("Unsupported functionality")
+
+
+def _calculate_values(function, function_space, points):
+    "Calculate function values at given points"
```

Comment:
I think you can remove the need to pass the `function_space` argument because it's just available as `function.function_space()`.